src/funcs.c

```diff
@@ -2825,3 +2825,71 @@ void joy_right()
   set(app->LV_GamesList, MUIA_List_Active, ind-1);
 
 }
+
+ULONG get_wb_version()
+{
+	static ULONG ver = 0UL;
+
+	if (ver != 0UL)
+	{
+		return ver;
+	}
+
+	if (WorkbenchBase == NULL)
+	{
+		//Really Workbench library should be already opened
+		// Somehow we're running without Workbench.
+		// Nothing to do since the version variable inits with 0.
+		return ver;
+	}
+
+	// Save workbench.library version for later calls
+	ver = WorkbenchBase->lib_Version;
+
+	return ver;
+}
+
+void open_current_dir()
+{
+	if (get_wb_version() < 44)
+	{
+		// workbench.library doesn't support OpenWorkbenchObjectA yet
+		return;
+	}
+
+	// Allocate Memory for variables
+	char* game_title = NULL;
+
+	//set the elements on the window
+	DoMethod(app->LV_GamesList, MUIM_List_GetEntry, MUIV_List_GetEntry_Active, &game_title);
+	if (game_title == NULL || strlen(game_title) == 0)
+	{
+		msg_box((const char*)GetMBString(MSG_SelectGameFromList));
+		return;
+	}
+
+	const int found = title_exists(game_title);
+	if (!found)
+	{
+		msg_box((const char*)GetMBString(MSG_SelectGameFromList));
+		return;
+	}
+	
+	int full_path_length;
+	int name_length;
+	int path_length;
+	full_path_length = strlen(item_games->path);
+	name_length = strlen(PathPart(item_games->path));
+	path_length = (full_path_length - name_length)+1;
+	
+	char *path_only = AllocVec(path_length, MEMF_ANY);		//reserve memory for path
+	if(!path_only) Printf("no memory?\n");
+	strncpy(path_only,item_games->path, path_length-1);		//Copy only the path w/o filename
+	path_only[path_length-1] = '\0';						//add NULL terminator
+	
+	
+	//Open path directory
+	OpenWorkbenchObject(path_only);
+	FreeVec(path_only);
+
+}
```

Comment:
I would prefer to move the logic to get the path to an other method, get_path, which will return it, if possible, and be called by this method. This way that new method will be able to be used in other places if needed.

Reply:
There's already `get_directory_name()` further up in funcs.c, so there must be reason why this was written with its own code to get the game directory. @T0lk13n, could you explain the decision here?

Reply:
In my opinion, it would be better to use that method, if possible, or change it slightly to get the results we need. It would be better to keep methods as small as possible, doing a single thing, if possible. This way they are going to be more reusable. 

That's of' course my opinion only. It's up to the owner to decide how he wants to deal with that.

Reply:
I guess I did it in that matter cos I didnt know igame code very well.
You can change all you want to make it cleaner.

Reply:
Will do!

Reply:
Turns out that `get_directory_name()` gets _just_ the name of the directory in the provided path string, not the full directory path. I'm adding a new `get_directory_path()` function that gives us what we actually want.